core/sparse_array.cpp

```diff
@@ -145,26 +145,13 @@ void SparseArray::Resize(uint32_t p_row)
 {
     //vectors are reallocated by 1000 entries at a time. It can be wasteful but probably better than making a function call for each new entry. Will defer the final decision to Ben. 
 
+    *(nnz_capacity + p_row)= (*(nnz_capacity + p_row)==0) ? initial_width : *(nnz_capacity + p_row) * 2; //Initialize or double the capacity 
 
-    *(columns + p_row)=(uint32_t *)realloc(*(columns + p_row), 1000 * sizeof(uint32_t));  
-    *(distances + p_row)=(uint32_t *)realloc(*(distances + p_row), 1000 * sizeof(uint32_t));
+    *(columns + p_row)=(uint32_t *)realloc(*(columns + p_row), *(nnz_capacity + p_row) * sizeof(uint32_t));  
+    *(distances + p_row)=(sa_distance_t *)realloc(*(distances + p_row), *(nnz_capacity + p_row) * sizeof(sa_distance_t));
```

Comment:
I missed the fact that distances had their own struct and initialized with int. Took care of that.